src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/MessageboxInstancesController.cs

```diff
@@ -119,5 +120,54 @@ public async Task<ActionResult> GetMessageBoxInstance(int instanceOwnerId, Guid
 
             return Ok(messageBoxInstance);
         }
+
+        /// <summary>
+        /// Undelete a soft deleted instance
+        /// </summary>
+        /// <param name="instanceOwnerId">instance owner</param>
+        /// <param name="instanceGuid">instance id</param>
+        /// <returns>True if the instance was undeleted.</returns>
+        [HttpPut("{instanceOwnerId:int}/{instanceGuid:guid}/undelete")]
+        public async Task<ActionResult> Undelete(int instanceOwnerId, Guid instanceGuid)
+        {
+            string instanceId = $"{instanceOwnerId}/{instanceGuid}";
+
+            Instance instance;
+
+            try
+            {
+                instance = await _instanceRepository.GetOne(instanceId, instanceOwnerId);
+            }
+            catch (DocumentClientException dce)
+            {
+                if (dce.Error.Code.Equals("NotFound"))
+                {
+                    return NotFound($"Didn't find the object that should be restored with instanceId={instanceId}");
+                }
+
+                return StatusCode(500, $"Unknown database exception in restore: {dce}");
```

Comment:
why should we return as unknown exception when we know the exception?

Reply:
I followed the standard of the other storage error messages. The exception class contains multiple error codes, but we could just return. "Database exception in restore: {ex}" ?